### DISCUSSION TITLE
Using the udp2raw as a docker

### DISCUSSION THREAD

**Discussion by raminrajabi** [wangyu-/udp2raw-tunnel#481] [General]
Hi, 
I'm using the udp2raw docker from the link below inside my Mikrotik device and I think it's old and not up-to-date and when I tried to make the docker image from the Dockerfile in the project it's giving me the compile error. Can anyone help me about this.
Thanks.
https://hub.docker.com/r/wangyucn/udp2raw-tunnel

**Comment by raminrajabi** [comment]
I managed to compile the image but now it's not working in the Mikrotik container:


FROM alpine:3.6 as builder

WORKDIR /

RUN apk add --no-cache git  build-base linux-headers && \
 git clone https://github.com/wangyu-/udp2raw-tunnel.git  && \
 cd udp2raw-tunnel && \
 make dynamic

FROM alpine:3.6
RUN apk add --no-cache libstdc++ iptables
COPY --from=builder /udp2raw-tunnel/udp2raw_dynamic /bin/
ENTRYPOINT [ "/bin/udp2raw_dynamic" ]



-----------------------------------------------------------------
[0mmodprobe: can't change directory to '/lib/modules': No such file or directory

iptables v1.6.1: can't initialize iptables table `filter': Table does not exist (do you need to insmod?)

**Comment by wangyu-** [comment]
docker in general doesn't allow raw socket to passthrough well, unless on linux + correct setting
udp2raw in general doesn't work well with docker.

>I think it's old and not up-to-date

Yes you are right.  I am considering deleting that docker repo to avoide the confusion,
